tests/ballerina-integration-test/src/test/resources/websub/advanced_services/01_websub_publisher.bal

```diff
@@ -23,6 +23,25 @@ import ballerina/websub;
 const string WEBSUB_PERSISTENCE_TOPIC_ONE = "http://one.persistence.topic.com";
 const string WEBSUB_PERSISTENCE_TOPIC_TWO = "http://two.persistence.topic.com";
 
+http:AuthProvider basicAuthProvider = {
+    scheme: "basic",
+    authStoreProvider: "config"
+};
+
+http:ServiceEndpointConfiguration hubListenerConfig = {
+    authProviders: [basicAuthProvider],
+    secureSocket: {
+        keyStore: {
+            path: "${ballerina.home}/bre/security/ballerinaKeystore.p12",
+            password: "ballerina"
+        },
+        trustStore: {
+            path: "${ballerina.home}/bre/security/ballerinaTruststore.p12",
+            password: "ballerina"
+        }
+    }
+};
+
```

Comment:
We should also add a test for publishing via the hub client when auth is enabled right? Shall we create an issue, if not possible with this PR?

Reply:
Added a test case